File: samples/scala/shorturls/app/controllers/ShortURLController.scala
package controllers

import org.reactivecouchbase.play.crud.CouchbaseCrudSourceController
import models.ShortURL
import models.ShortURLs.fmt
import org.reactivecouchbase.play.PlayCouchbase
import play.api.Play.current

object ShortURLController extends CouchbaseCrudSourceController[ShortURL] {
  def bucket = PlayCouchbase.bucket("default")
  override def defaultViewName = "by_url"
  override def defaultDesignDocname = "shorturls"
  override def idKey = "id"
}

File: samples/java/shorturls/app/controllers/ShortURLController.java
package controllers;

import models.ShortURL;

import org.reactivecouchbase.play.java.Couchbase;
import org.reactivecouchbase.play.java.crud.CrudSource;
import org.reactivecouchbase.play.java.crud.CrudSourceController;

public class ShortURLController extends CrudSourceController<ShortURL> {

    private final CrudSource<ShortURL> source = new CrudSource<ShortURL>( Couchbase.bucket("default"), ShortURL.class );

    @Override
    public CrudSource<ShortURL> getSource() {
        return source;
    }

    @Override
    public String defaultDesignDocname() {
        return "shorturls";
    }

    @Override
    public String defaultViewName() {
        return "by_url";
    }
}
